packages/checkout-storefront/src/hooks/useUser.ts

```diff
@@ -0,0 +1,12 @@
+import { useUserQuery } from "@/checkout-storefront/graphql";
+import { useMemo } from "react";
+
+export const useUser = () => {
+  const [{ data, fetching: loading, stale }] = useUserQuery();
+
+  const user = data?.user;
+
+  const authenticated = useMemo(() => !!user?.id, [user?.id]);
```

Comment:
This doesn't need `useMemo`:
```suggestion
  const authenticated = !!user?.id;
```